thorlcr/slave/slwatchdog.cpp

```diff
@@ -98,7 +98,9 @@ class CGraphProgressHandlerBase : public CSimpleInterface, implements ISlaveWatc
                 ForEachItemIn(g, activeGraphs) // NB: 1 for each slavesPerProcess
                 {
                     CGraphBase &graph = activeGraphs.item(g);
-                    graph.serializeStats(progressData);
+                    progressData.append((unsigned)graph.queryJobChannel().queryMyRank()-1);
+                    if (!graph.serializeStats(progressData))
```

Comment:
@jakesmith can you explain why this fixes it?  It isn't clear to me what difference this would have